common/scheduledevents2/backgroundworker.go

```diff
@@ -104,7 +104,7 @@ func runFlushNewEvents() error {
 // flushEventToRedis flushes an event to redis, this is done as a performance improvement as the postgres db is only queried as often
 func flushEventToRedis(c radix.Client, evt *models.ScheduledEvent) error {
 	v := fmt.Sprintf("%d:%d", evt.ID, evt.GuildID)
-	err := c.Do(radix.Cmd(nil, "ZADD", "scheduled_events_soon", strconv.FormatInt(evt.TriggersAt.UTC().Unix(), 10), v))
+	err := c.Do(radix.Cmd(nil, "ZADD", "scheduled_events_soon", strconv.FormatInt(evt.TriggersAt.UTC().UnixMicro(), 10), v))
```

Comment:
Out of curiosity, what's up with this change?

Reply:
Since, `.Unix()` does not consider microseconds, the events were being triggered a few microseconds before they were supposed to. This change allows the events to trigger more precisely.
(The reason for updating this is that some events were being re-scheduled, because there was still some microseconds left before they were actually supposed to get triggered)

Reply:
Interesting! Seems like a reasonable change. I was thinking about what occurs when there are entries in `scheduled_events_soon` with second-precision scores (i.e., added before this change) in addition to microsecond-precision scores (i.e., after this change) and whether this would lead to issues due to the different format, but it turns out the existing code should handle this just fine. Nonetheless, perhaps testing such a scenario might be prudent (to make sure events added before still work), if you haven't done so already.

Reply:
Just after deployment, the entries in `scheduled_events_soon` will be scheduled (because their score will be less than the time generated by UnixMicro), but it will be a one time thing and yag can handle it elegantly.